### PR TITLE
Set proper value when upgrading app patch version

### LINKED ISSUE #34851: App version update issue
### Steps

1. Checkout out OC master
1. Change info.xml of "files" app to say "1.5.3"
1. Open web browser, this would run any query
1. Check `select * from oc_appconfig where configkey='installed_version' and appid='files';`

### Expected result
Version properly updated.

### Actual result
Version not updated.

Turns out the code is setting the old value repeatedly, and this happens for every single request!

### PR DESCRIPTION
**Pull request by PVince81**
## Related Issue
Fixes https://github.com/owncloud/core/issues/34851

## How Has This Been Tested?
Manual test with steps
- [ ] TODO: add unit test if possible

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/lib/legacy/AppTest.php [hunk 1]
```diff
@@ -22,15 +22,30 @@
...
 use OCP\App\AppNotFoundException;
-use function Test\AppFramework\rrmdir;
 use Test\TestCase;
+use OCP\App\IAppManager;
+use OCP\IAppConfig;
 
 class AppTest extends TestCase {
 	private $appPath;
 
+	private static function rrmdir($directory) {
```
Comment (PVince81) [inline tests/lib/legacy/AppTest.php]: moved from AppFramework and adjusted

**Review by IljaN — APPROVED** [review]

**Comment by PVince81** [comment]
Damn, seems that my little hack that actually clears the app cache is now breaking other tests that were somehow relying on it...

**Comment by PVince81** [comment]
I've removed the unit tests, cannot make it work without destroying many other tests due to side effects.

we should look into testing such app update consistencies as part of update testing.
raised https://github.com/owncloud/QA/issues/613

I'll merge and backport

**Comment by PVince81** [comment]
stable10: https://github.com/owncloud/core/pull/34859

**Pull request merged**